src/main/java/org/sonarsource/sonarlint/ls/backend/BackendServiceFacade.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * SonarLint Language Server
+ * Copyright (C) 2009-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.ls.backend;
+
+import java.util.Map;
+import org.sonarsource.sonarlint.core.clientapi.SonarLintBackend;
+import org.sonarsource.sonarlint.core.clientapi.backend.InitializeParams;
+import org.sonarsource.sonarlint.ls.settings.ServerConnectionSettings;
+
+public class BackendServiceFacade {
+
+  private final BackendService backend;
+  private final BackendInitParams initParams;
+  private boolean initialized = false;
```

Comment:
Is the facade used by different threads ? You might need to use an atomic reference to ensure all threads access the same value.